### PR TITLE
[Bounce] [Reviewed] Add "Bounce Count" property (and simplified wording)

### PR DESCRIPTION
**Pull request by tristanbob**
## Playable game
https://gd.games/victrisgames/common-hair--breakout-

## Project files
[BounceCount.zip](https://github.com/GDevelopApp/GDevelop-extensions/files/11186270/BounceCount.zip)

## Video
https://user-images.githubusercontent.com/8879811/230795866-68270ecb-ef07-4033-91c8-9280b5e1d122.mp4

### CODE REVIEW

File: extensions/reviewed/Bounce.json [hunk 1]
```diff
@@ -2,46 +2,47 @@
...
+  "shortDescription": "Make an object bounce away from another object",
+  "version": "0.1.0",
   "description": [
-    "Provides an action to make the object bounce from another object it just touched.",
+    "Provides an action to make the object bounce away from another object it just touched. A permanent force is applied to the object when it is in collision with another object, which makes it bounce realistically.",
     "",
-    "> ⚠️ This *won't work* with Physics engine or Platformer characters. This is working with objects that are moved using actions or behaviors using **the forces**.",
+    "*Bounce count* can be used to trigger actions after the object bounces a certain number of times, such as bullets that ricochet off a few walls.",
     "",
-    "The Breakout-like example uses this extension to make balls bounce ([open the project online](https://editor.gdevelop.io/?project=example://breakout))."
```
Comment (D8H) [inline extensions/reviewed/Bounce.json:17]: This should probably not be removed.

File: extensions/reviewed/Bounce.json [hunk 2]
```diff
@@ -2,46 +2,47 @@
...
-      "description": "Provides an action to make the object bounce from another object it just touched. Add a permanent force to the object and, when in collision with another one, use the action to make it bounce realistically.",
+      "description": "Make an object bounce away from another object.",
       "fullName": "Bounce",
       "name": "Bounce",
       "objectType": "",
       "eventsFunctions": [
         {
-          "description": "Bounce the object off another object it is currently colliding with, according to the angle and the speed of forces applied on the object.\nMake sure to test for a collision between the two objects before launching this action. All the forces will be removed from the object, and a new permanent force will be added to make the object bounce.",
-          "fullName": "Bounce off another object",
+          "description": "Bounce the object away from another object. Use a condition to check when the objects are colliding.",
```
Comment (D8H) [inline extensions/reviewed/Bounce.json:41]: It should remind that this action works only with forces.

File: extensions/reviewed/Bounce.json [hunk 3]
```diff
@@ -178,172 +179,15 @@
...
-            }
-          ],
-          "objectGroups": []
-        },
-        {
-          "description": "Bounce the object off another object it is currently colliding with, according to the angle and the speed of forces applied on the object.\nThe bounce will always be calculated *to go toward the specified angle (the \"normal angle\")*. For example, if the object is arriving at this exact angle, it will bounce in the opposite direction.\n\nMake sure to test for a collision between the two objects before launching this action. All the forces will be removed from the object, and a new permanent force will be added to make the object bounce.",
-          "fullName": "Bounce off another object toward a specified angle",
-          "functionType": "Action",
-          "name": "BounceOffSpecificAngle",
-          "sentence": "Bounce _PARAM0_ off _PARAM2_ assuming a normal angle of _PARAM3_",
```
Comment (D8H) [inline extensions/reviewed/Bounce.json:212]: Why removing this?

File: extensions/reviewed/Bounce.json [hunk 4]
```diff
@@ -365,91 +209,39 @@
...
-            }
-          ],
-          "objectGroups": []
-        },
-        {
-          "description": "Bounce the object off another object it is currently colliding with, according to the angle and the speed of forces applied on the object.\nThe bounce will always be vertical, like if the object is *colliding a perfectly horizontal obstacle* (like the top/bottom of the screen in a pong game). For example, if the object is arriving with an angle of exactly 90 degrees, it will bounce in the opposite direction: -90 degrees.\n\nMake sure to test for a collision between the two objects before launching this action. All the forces will be removed from the object, and a new permanent force will be added to make the object bounce.",
-          "fullName": "Bounce vertically",
-          "functionType": "Action",
-          "name": "BounceOffVertically",
-          "sentence": "Bounce _PARAM0_ off _PARAM2_ - always vertically",
```
Comment (D8H) [inline extensions/reviewed/Bounce.json:386]: Why removing this?

**Comment by D8H** [comment]
Are you sure you used the last version of the extension?

**Comment by tristanbob** [comment]
> Are you sure you used the last version of the extension?

Good catch.  I made the mistake (again) of starting from the version in the game example repo.  

I have imported the current extension and then applied my changes.

File: extensions/reviewed/Bounce.json [hunk 5]
```diff
@@ -469,6 +492,47 @@
...
           ],
           "objectGroups": []
+        },
+        {
+          "description": "the number of times this object has bounced off another object of the object.",
+          "fullName": "Number of times this object has bounced off another object",
+          "functionType": "ExpressionAndCondition",
+          "group": "Bounce configuration",
+          "name": "BounceCount",
+          "sentence": "the number of times this object has bounced off another object",
```
Comment (D8H) [inline extensions/reviewed/Bounce.json:502]: How does the condition look in an event?

File: extensions/reviewed/Bounce.json [hunk 6]
```diff
@@ -469,6 +492,47 @@
             }
           ],
           "objectGroups": []
+        },
+        {
+          "description": "the number of times this object has bounced off another object of the object.",
+          "fullName": "Number of times this object has bounced off another object",
```
Comment (D8H) [inline extensions/reviewed/Bounce.json:498]: The name must be as short as possible to allow user to scan them.
```suggestion
          "fullName": "Bounce count",
```

File: extensions/reviewed/Bounce.json [hunk 7]
```diff
@@ -469,6 +492,47 @@
...
+          "functionType": "ExpressionAndCondition",
+          "group": "Bounce configuration",
```
Comment (D8H) [inline extensions/reviewed/Bounce.json:500]: It's a state not a configuration.
```suggestion
          "group": "",
```

File: extensions/reviewed/Bounce.json [hunk 8]
```diff
@@ -521,6 +585,17 @@
           "extraInformation": [],
           "hidden": true,
           "name": "NormalAngle"
+        },
+        {
+          "value": "0",
+          "type": "Number",
+          "unit": "Dimensionless",
+          "label": "Number of times this object has bounced off another object",
+          "description": "",
```
Comment (D8H) [inline extensions/reviewed/Bounce.json:594]:
```suggestion
          "label": "Bounce count",
          "description": "Number of times this object has bounced off another object",
```

File: extensions/reviewed/Bounce.json (hunk 5 above)
Comment (D8H) [reply extensions/reviewed/Bounce.json:502]:
```suggestion
          "sentence": "the bounce count",
```

**Comment by tristanbob** [comment]
> Why did you close it?

I have no idea, it was not intentional. I have reopened this PR.

File: extensions/reviewed/Bounce.json [hunk 9]
```diff
@@ -469,6 +492,47 @@
             }
           ],
           "objectGroups": []
+        },
+        {
+          "description": "the number of times this object has bounced off another object of the object.",
```
Comment (D8H) [inline extensions/reviewed/Bounce.json:497]:
```suggestion
          "description": "the number of times this object has bounced off another object.",
```

**Review by D8H — APPROVED** [review]

**Pull request merged**